lib/smart_proxy_dynflow_core/api.rb

```diff
@@ -7,13 +7,14 @@ class Api < ::Sinatra::Base
 
     before do
       logger = Log.instance
-      authorize_with_ssl_client
+      authorize_with_token || authorize_with_ssl_client
       content_type :json
     end
 
     post "/tasks/?" do
       params = MultiJson.load(request.body.read)
-      trigger_task(::Dynflow::Utils.constantize(params['action_name']), params['action_input']).to_json
+      trigger_task(::Dynflow::Utils.constantize(params['action_name']),
+                   params['action_input'].merge(:callback_host => request.env['HTTP_X_FORWARDED_FOR'])).to_json
```

Comment:
Note: `X-Forwarded-For` won't be set if dynflow_core is not a standalone process, will have to fallback to `HTTP_HOST` in that case